Capri.Services/Faculties/FacultyCreator.cs

```diff
@@ -0,0 +1,41 @@
+using System.Threading.Tasks;
+using AutoMapper;
+using Capri.Database;
+using Capri.Database.Entities;
+using Capri.Web.ViewModels.Faculty;
+
+namespace Capri.Services.Faculties
+{
+    public class FacultyCreator : IFacultyCreator
+    {
+        private readonly ISqlDbContext _context;
+        private readonly IMapper _mapper;
+
+        public FacultyCreator(
+            ISqlDbContext context,
+            IMapper mapper
+        )
+        {
+            _context = context;
+            _mapper = mapper;
+        }
+
+        public async Task<IServiceResult<FacultyViewModel>> Create(FacultyRegistration registration)
+        {
+            var faculty = _mapper.Map<Faculty>(registration);
+
+            try 
+            {
+                await _context.Faculties.AddAsync(faculty);
+                await _context.SaveChangesAsync();
+            }
+            catch
```

Comment:
Same as above.

Reply:
I expect invalid data. Unless we validated data before calling this method

Reply:
I expect invalid data. Unless we validated data before calling this method